org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/HoverInfoProvider.java

```diff
@@ -121,24 +123,33 @@ private boolean isResolved(IJavaElement element, IProgressMonitor monitor) throw
 		res[0] = false;
 		SearchEngine engine = new SearchEngine();
 		IJavaSearchScope scope = SearchEngine.createJavaSearchScope(new IJavaElement[] { unit }, IJavaSearchScope.SOURCES);
-		engine.search(pattern, new SearchParticipant[] { SearchEngine.getDefaultSearchParticipant() }, scope, new SearchRequestor() {
+		try {
+			engine.search(pattern, new SearchParticipant[] { SearchEngine.getDefaultSearchParticipant() }, scope, new SearchRequestor() {
 
-			@Override
-			public void acceptSearchMatch(SearchMatch match) throws CoreException {
-				if (match.getAccuracy() == SearchMatch.A_INACCURATE) {
-					return;
-				}
-				Object o = match.getElement();
-				if (o instanceof IJavaElement) {
-					IJavaElement element = (IJavaElement) o;
-					ICompilationUnit compilationUnit = (ICompilationUnit) element.getAncestor(IJavaElement.COMPILATION_UNIT);
-					if (compilationUnit == null) {
+				@Override
+				public void acceptSearchMatch(SearchMatch match) throws CoreException {
+					if (match.getAccuracy() == SearchMatch.A_INACCURATE) {
 						return;
 					}
-					res[0] = true;
+					Object o = match.getElement();
+					if (o instanceof IJavaElement) {
+						IJavaElement element = (IJavaElement) o;
+						ICompilationUnit compilationUnit = (ICompilationUnit) element.getAncestor(IJavaElement.COMPILATION_UNIT);
+						if (compilationUnit == null) {
+							return;
+						}
+						res[0] = true;
+						IStatus status = new Status(IStatus.OK, JavaLanguageServerPlugin.PLUGIN_ID, "OK");
+						throw new CoreException(status);
```

Comment:
when using exceptions for control flow (which is an antipattern), you'll want to avoid computing the stacktrace (which is slow), so you need to override it like 
```java
    @Override
    public synchronized Throwable fillInStackTrace() {
        return this;
    }
```